### ISSUE TITLE
[feature]: combine [ActionController] with [ScrollController] to control the slidable panel centrally

### ISSUE THREAD

**Issue by SimonWang9610** [SimonWang9610/flutter_slidable_panel#1]
Currently, users must create and use `ScrollController` and `ActionController` separately, like:
- create controllers
```dart
  final SlideController _slideController = SlideController();
  final ActionController _actionController = ActionController();
```
Particularly, users must manually reset `ActionController` when dismissing the slidable panel; otherwise, the slidable panel would keep the previous state changed by `ActionController` when opening the panel again.
- use controllers
```dart
_slideController.dismiss(
   onDismissed: () {
       _actionController.reset();
   }
);
```

**Comment by SimonWang9610** [comment]
Alos, users must bind the `ActionController` to `SlideActionPanel` manually, which is error-prone, like:
```dart
preActionPanel: SlideActionPanel(
            actionLayout: ActionLayout.spaceEvenly(ActionMotion.scroll),
            slidePercent: _slideController.slidePercent,

            /// bind [ActionController] with the [SlideActionPanel]
            controller: _actionController,
            actions: [
              TextButton(
                onPressed: () {
                  _actionController.toggle(0);
                },
                style: TextButton.styleFrom(
                  backgroundColor: Colors.greenAccent,
                  shape: const RoundedRectangleBorder(),
                ),
                child: const Text("Archive"),
              ),
              TextButton(
                onPressed: () {
                  _actionController.toggle(1);
                },
                style: TextButton.styleFrom(
                  backgroundColor: Colors.redAccent,
                  shape: const RoundedRectangleBorder(),
                ),
                child: const Text("Delete"),
              ),
            ],
          ),
```

**Comment by SimonWang9610** [comment]
Hopefully, users should only use a single `SlideController` to control sliding and expanding.

**Comment by SimonWang9610** [comment]
fix in 6fd3f32dd4dd547915efc92bce1923033d5b04ea
